app/controllers/patients_controller.rb

```diff
@@ -465,18 +463,16 @@ def update_status
       end
     end
 
-    # Update patient and all group members
-    if params.permit(:apply_to_household)[:apply_to_household]
-      ([patient] + (current_user.get_patient(patient.responder_id)&.dependents || [])).uniq.each do |member|
-        update_monitoring_fields(member, params, patient[:id] == member[:id] ? :patient : :dependent, :group)
-      end
-      # Update patient and all group members in continuous exposure
-    elsif params.permit(:apply_to_household_cm_only)[:apply_to_household_cm_only] # update patient and group members only with continuous exposure on
-      ([patient] + (current_user.get_patient(patient.responder_id)&.dependents&.where(continuous_exposure: true) || [])).uniq.each do |member|
-        update_monitoring_fields(member, params, patient[:id] == member[:id] ? :patient : :dependent, :group_cm)
-      end
-    else # Update patient
-      update_monitoring_fields(patient, params, :patient, :none)
+    # Update patient
+    update_monitoring_fields(patient, params, :patient, :none)
+
+    # If not applying to household, return
+    return unless params.permit(:apply_to_household)[:apply_to_household] && params[:apply_to_household_ids].count.positive?
+
+    # Update selected group members if applying to household and ids are supplied
+    params[:apply_to_household_ids].each do |id|
+      member = current_user.get_patient(id)
```

Comment:
There is a chance member could be nil. Make sure either `update_monitoring_fields` can handle that, or you move to the next loop here.

Reply:
updated here: https://github.com/SaraAlert/SaraAlert/pull/722/commits/9d58d8d4f855e244bb449310e65e3d92a4657f35